File: logolicious/src/main/java/com/olav/logolicious/screens/fragments/Fragment_AboutInfo.java
package com.olav.logolicious.screens.fragments;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.CompoundButton.OnCheckedChangeListener;
import android.widget.TextView;

import com.olav.logolicious.R;
import com.olav.logolicious.util.GlobalClass;

public class Fragment_AboutInfo extends Fragment {

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		View rootView = inflater.inflate(R.layout.about_info_credits, container, false);
		setHasOptionsMenu(false);
		final TextView version = (TextView) rootView.findViewById(R.id.version);
		final CheckBox showTipsNextime = (CheckBox) rootView.findViewById(R.id.showTipNextime);
		PackageInfo packageInfo = null;
		try {
			packageInfo = getActivity().getPackageManager().getPackageInfo(getActivity().getPackageName(), 0);
			version.setText("Version " + packageInfo.versionName);
		} catch (PackageManager.NameNotFoundException e2) {
			version.setText("Version is up to date");
		}
		
		if (GlobalClass.sqLiteHelper.isShowHint() == 1) {
			showTipsNextime.setChecked(true);
		} else {
			showTipsNextime.setChecked(false);
		}

		showTipsNextime.setOnCheckedChangeListener(new OnCheckedChangeListener() {

			@Override
			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
				// TODO Auto-generated method stub
					if (isChecked) {
						GlobalClass.sqLiteHelper.updateHint(1,1);
					} else {
						GlobalClass.sqLiteHelper.updateHint(1,0);
					}	
			}
		});
		
		
		return rootView;
	}

}

File: logolicious/src/main/java/com/olav/logolicious/screens/fragments/FragmentAboutInfo.kt
package com.olav.logolicious.screens.fragments

import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.olav.logolicious.R
import com.olav.logolicious.util.GlobalClass

class FragmentAboutInfo : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val rootView = inflater.inflate(R.layout.about_info_credits, container, false)
        setHasOptionsMenu(false)
        val version = rootView.findViewById<View>(R.id.version) as TextView
        val showTipsNextime = rootView.findViewById<View>(R.id.showTipNextime) as CheckBox
        var packageInfo: PackageInfo? = null
        try {
            packageInfo = activity!!.packageManager.getPackageInfo(activity!!.packageName, 0)
            version.text = "Version " + packageInfo.versionName
        } catch (e2: PackageManager.NameNotFoundException) {
            version.text = "Version is up to date"
        }
        showTipsNextime.isChecked = GlobalClass.sqLiteHelper.isShowHint == 1
        showTipsNextime.setOnCheckedChangeListener { _, isChecked ->
            if (isChecked) {
                GlobalClass.sqLiteHelper.updateHint(1, 1)
            } else {
                GlobalClass.sqLiteHelper.updateHint(1, 0)
            }
        }
        return rootView
    }
}